maven_iuvs/geometry.py

```diff
@@ -941,3 +941,178 @@ def pixelcorner_avg(pixel_x, pixel_y, pixel_z=None,
         return avgxy[:, :, 0], avgxy[:, :, 1], avgxy[:, :, 2]
 
     return avgxy[:, :, 0], avgxy[:, :, 1]
+
+
+def has_geometry_pvec(hdul):
+    """
+    Determines whether geodetic latitudes are available for the pixels in the pixel vector
+
+    Parameters
+    ----------
+    hdul : astropy FITS HDUList object
+           HDU list for a given observation
+
+    Returns
+    -------
+    n_int : int
+            number of integrations
+
+    """    
+    geom_quantity = hdul['PixelGeometry'].data['PIXEL_CORNER_LAT']
+    
+    n_nan = np.sum(np.isnan(geom_quantity))
+    n_quant = np.product(np.shape(geom_quantity))
+
+    nanfrac = n_nan / n_quant
+    
+    return (nanfrac < 1.0)
+
+
+def find_files_missing_geometry(file_index, show_total=False):
+    """
+    Identifies observation files with geometry
+
+    Parameters
+    ----------
+    file_index : index file (.npy) 
+                 dictionaries containing metadata of various observation files
+    show_total: binary
+                whether to print what fraction of total the missing files are
+    Returns
+    ----------
+    no_geom: list
+             metadata for files with don't have geometry
+    """
+    no_geom = [f for f in file_index if 'orbit' in f['name'] and not f['geom']]
+    
+    if show_total==True:
+        all_orbit_files = [f for f in file_index if 'orbit' in f['name']]
+        print(f'{len(no_geom)} of {len(all_orbit_files)} have no geometry.\n')
+        
+    return no_geom
+
+
+def find_files_with_geometry(file_index):
+    """
+    Opposite of find_files_missing_geometry
+
+    Parameters
+    ----------
+    file_index : index file (.npy) 
+                 dictionaries containing metadata of various observation files
+    show_total: binary
+                whether to print how many files of the total the files missing geometry comprise
+    Returns
+    ----------
+    with_geom: list
+             metadata for files with don't have geometry
+    """
+    with_geom = [f for f in file_index if 'orbit' in f['name'] and f['geom']]
+
+    # print(f'{len(with_geom)} have geometry.\n')
+    return with_geom
+
+
+def make_sza_plot(ax, fitfile, linecolor="cornflowerblue"):
```

Comment:
move to graphics

Reply:
Fixed by c361af4d4d7026605a880d550706d37d7c4378c2